### PR TITLE
User story 9

### LINKED ISSUE #22: User Story 9, Book Index Page shows Statistics
```
As a Visitor,
As a Visitor,
When I visit the book index page,
I see an area showing statistics about all books:
- the three highest-rated books (book title and average rating)
- thre three worst-rated books (book title and average rating)
- three users who have written the most reviews (user name and review count)
```

DO THESE STORIES FIRST:
- User Story 6, Book Index Page

### Checklist:

- [x] base code tests are written
- [x] base code is written to pass tests
- [x] edge case ("sad path") tests written
- [x] edge case code written
- [x] code is reviewed
- [x] branch is merged
- [x] all tests still pass

child of #26

child of #33

depends on #25

### PR DESCRIPTION
**Pull request by chakeresa**
This PR/branch...
 - Resolves #22 (User Story 9 - book index page statistics)
 - Shows the 3 books with the highest & lowest average ratings (but does not include books that haven't been rated)
 - Shows the 3 users with the most reviews written (but does not include users that haven't written any reviews)
 - Adds some class methods to Book & User to calculate the stats above
 - Tests all of the above
 - Adds some more seed data so the statistics on the development/production sites are more interesting

### CODE REVIEW

File: spec/features/books/index_statistics_spec.rb [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    it "shows the 3 highest rated books" do
+      visit books_path
+
+      within "#statistics" do
+        expect(page).to have_content("Statistics")
+
+        within "#highest-rated" do
+          expect(page).to have_content("Highest Rated Books")
+          expect(page.all("li")[0]).to have_content(@book_5.title)
```
Comment (DavisC0801) [inline spec/features/books/index_statistics_spec.rb]: This block should be changed to have_links for book titles instead.

File: spec/models/book_spec.rb [hunk 2]
```diff
@@ -101,6 +101,45 @@
...
+        expect(Book.highest_rated(3)).to eq([@book_5, @book_2, @book_4])
+        expect(Book.highest_rated(1)).to eq([@book_5])
+      end
+
+      it "finds the X lowest rated books" do
+        expect(Book.lowest_rated(5)).to eq([@book_1, @book_4, @book_2, @book_5])
+        # books with no reviews (e.g. @book_3) are not included
+        expect(Book.lowest_rated(3)).to eq([@book_1, @book_4, @book_2])
+        expect(Book.lowest_rated(1)).to eq([@book_1])
+      end
```
Comment (DavisC0801) [inline spec/models/book_spec.rb:141]: Sad path test - Book.lowest_rated and highest_rated with no book records.

File: spec/features/books/index_statistics_spec.rb [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      visit books_path
+
+      within "#statistics" do
+        expect(page).to have_content("No books have reviews yet.")
+        expect(page).to_not have_content("Highest Rated Books")
+        expect(page).to_not have_content("Highest Rated Books")
+        expect(page).to_not have_content("Users with Most Reviews Written")
+      end
+
+      save_and_open_page
```
Comment (DavisC0801) [inline spec/features/books/index_statistics_spec.rb]:
```suggestion
      save_and_open_page
```

This line should be removed.

**Review by DavisC0801 — COMMENTED** [review]
As always your work is really solid.

File: spec/models/book_spec.rb (hunk 2 above)
Comment (chakeresa) [reply spec/models/book_spec.rb:141]: Done

File: spec/features/books/index_statistics_spec.rb (hunk 1 above)
Comment (chakeresa) [reply spec/features/books/index_statistics_spec.rb]: Good catch -- done for book titles as well as usernames
